lib/cn/cn_tree.c

```diff
@@ -1665,14 +1867,16 @@ cn_comp_commit(struct cn_compaction_work *w)
             goto done;
     }
 
-    w->cw_err = spill ? cn_comp_commit_spill(w, kvsets) :
-                        cn_comp_commit_kvcompact(w, kvsets[0]);
-    if (ev(w->cw_err))
-        goto done;
+    if (split)
+        w->cw_err = cn_comp_update_split(w, kvsets, nodeidv);
+    else
+        spill ? cn_comp_update_spill(w, kvsets) : cn_comp_update_kvcompact(w, kvsets[0]);
```

Comment:
Prob use a switch on action here?  But you've converted from actions to bools, so maybe a little awkward bouncing between the two...

Reply:
Yeah, I converted it to bool as I was tired writing if (w->cw_action == CN_ACTION_...). But I see that a switch action is better here.. Let me look at it again.

Reply:
I made it a switch action here.. The mix of both bool and switch doesn't look odd to me..